lib/utilities/requestUtils.js

```diff
@@ -0,0 +1,35 @@
+const { config } = require('../Config');
+const { ipCheck } = require('arsenal');
+
+/**
+ * getClientIp - Gets the client IP from the request
+ * @param {object} request - http request object
+ * @param {object} s3config - (optional) s3 config
+ * @return {string} - returns client IP from the request
+ */
+function getClientIp(request, s3config) {
+    const clientIp = (request.connection && request.connection.remoteAddress) ||
+        (request.socket && request.socket.remoteAddress) ||
+        (request.connection && request.connection.socket &&
+            request.connection.socket.remoteAddress);
```

Comment:
In which case would `request.connection.socket` exist?

Reply:
I am not sure. `Vault` had this. I am reverting to use only `request.socket`.